src/models/StyledComponent.js

```diff
@@ -235,20 +234,22 @@ export default (ComponentStyle: Function, constructWithOptions: Function) => {
             ? rules
             : rulesFromOptions.concat(rules)
 
-        const newComponentId =
-          previousComponentId &&
-          `${previousComponentId}-${ComponentStyle.generateName(
-            newRules.join(''),
-          )}`
-
         const newOptions = {
           ...optionsToCopy,
-          componentId: newComponentId,
           rules: newRules,
           ParentComponent: StyledComponent,
         }
 
-        return constructWithOptions(createStyledComponent, target, newOptions)
+        const createWithHash = (tag, opts, css) => {
```

Comment:
This is not quite looking optimal, in terms of V8 preoptimisation. It will probably deopt this function, since it's created on every call. Any reason why you were unhappy with your previous implementation?